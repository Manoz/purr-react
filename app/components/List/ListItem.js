import styled from 'styled-components';

const ListItem = styled.li`
  font-size: 1.0625rem;
  line-height: 1.52947;
  font-weight: 400;
  letter-spacing: -.021em;
`;

export default ListItem;
