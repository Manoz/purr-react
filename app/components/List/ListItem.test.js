import React from 'react';
import { shallow } from 'enzyme';

import ListItem from './ListItem';

describe('<ListItem />', () => {
  it('should render a <li> tag', () => {
    const wrapper = shallow(<ListItem />);
    expect(wrapper.type()).toEqual('li');
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const wrapper = shallow(<ListItem id={id} />);
    expect(wrapper.prop('id')).toEqual(id);
  });

  it('should not adopt an invalid attribute', () => {
    const wrapper = shallow(<ListItem attribute="test" />);
    expect(wrapper.prop('attribute')).toBeUndefined();
  });
});
