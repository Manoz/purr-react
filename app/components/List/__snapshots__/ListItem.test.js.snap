// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> should render a <li> tag 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "ListItem-odeuvr-0",
        "isStatic": true,
        "rules": Array [
          "font-size:18px;line-height:1.45;font-weight:400;letter-spacing:-.021em;",
        ],
      },
      "displayName": "ListItem",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "ListItem-odeuvr-0",
      "target": "li",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
